activerecord/lib/active_record/database_configurations.rb

```diff
@@ -158,7 +158,7 @@ def env_with_configs(env = nil)
           configurations.select { |db_config| db_config.env_name == env }
         else
           configurations
-        end
+        end.sort_by!(&:migrations_priority)
```

Comment:
should this mutate the list?

```suggestion
        end.sort_by(&:migrations_priority)
```